pkg/unidling/controller/controller.go

```diff
@@ -92,9 +93,27 @@ func NewUnidlingController(scaleNS kextclient.ScalesGetter, endptsNS kclient.End
 
 	_, controller := framework.NewInformer(
 		&cache.ListWatch{
-			// No need to list -- we only care about new events
 			ListFunc: func(options kapi.ListOptions) (runtime.Object, error) {
-				return &kapi.EventList{}, nil
+				// Even though we don't care about current events and we only want to
+				// process new ones via the watch, we have to do a List() so we can get
+				// the current ResourceVersion. We'll return that in an empty EventList
+				// so that when the informer starts watching, it does it starting from
+				// the correct ResourceVersion. If we don't do this, the Kubernetes etcd
+				// watch code will still do a full list, and it will then attempt to
+				// Get() each item at revision <item.ResourceVersion - 1>. In the event
+				// that revision has been compacted, the entire watch will fail and the
+				// informer will be broken. (AG) I think this is a bug in the Kubernetes
+				// watch client code, but need to discuss this with the community.
+				list, err := evtNS.Events(kapi.NamespaceAll).List(options)
+				if err != nil {
+					return nil, err
+				}
+
+				return &kapi.EventList{
+					ListMeta: unversioned.ListMeta{
+						ResourceVersion: list.ResourceVersion,
```

Comment:
Since we've already paid the cost for the data (retrieval, deserialization, conversion), using it makes sense.

If we had a way to get a current resourceVersion and *not* pay those costs, I could see making the tradeoff.  We've had event spamming problems before and a full listing could end up problematic in those cases.